### PR TITLE
Add books field to GraphQL authors

### PR DESCRIPTION
**Pull request by hiterm**
## Summary

- add a non-null `books: [Book!]!` field to the GraphQL `Author` type
- batch book lookup by author IDs through the repository and query use case
- resolve author books with a request-scoped DataLoader to avoid N+1 queries
- preserve user scoping and return an empty list for authors without books
- add OpenSpec artifacts and repository, use-case, loader, schema, and database-backed GraphQL tests

## Impact

Clients can query books directly from one or more authors using the existing GraphQL `Book` type. The field has no pagination or ordering guarantee in this change.

## Validation

- `cargo fmt --check`
- `cargo clippy --all-targets --locked -- -D warnings`
- `cargo test --locked` (162 tests passed)
- `cargo test --locked --features test-with-database --no-run`

The database-backed tests compile successfully. Runtime execution was not available locally because `DATABASE_URL` is not configured in the environment.

## Summary by CodeRabbit

- **新機能**
  - GraphQLの`Author`型に、関連する書籍一覧を取得できる`books`フィールドを追加しました。
  - 書籍がない著者には空のリストを返します。
  - 複数著者に関連する書籍も、各著者の一覧に正しく表示されます。
  - 認証ユーザーが所有するデータのみを対象とし、データの分離を維持します。

- **テスト**
  - 複数著者、書籍なし、データ分離などのケースを検証するテストを追加しました。

### CODE REVIEW

**Comment by hiterm** [comment]
@codex review

**Comment by hiterm** [comment]
@codex review

File: openspec/changes/add-author-books-field/design.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+## Context
+
+The GraphQL `Book` type already resolves its authors with a request-scoped `AuthorLoader`, but the inverse relationship is not exposed. Resolving books independently for each author would create N+1 database queries. The existing repository and query-use-case layers use concrete generic types and tenant-scoped identifiers.
+
+## Goals / Non-Goals
+
+**Goals:**
+
+- Expose each author's books through the existing GraphQL `Book` type.
```
[the remark this answers is not in the document]
Comment (hiterm) [reply openspec/changes/add-author-books-field/design.md]: Codexとして対応しました。`e7b78d9` で、既存の GraphQL `Author` 型へ `books` を追加し、要素には既存の GraphQL `Book` 型を利用することが明確になるよう設計文書を修正しました。

File: src/use_case/interactor/query.rs [hunk 2]
```diff
@@ -310,6 +336,54 @@ mod tests {
...
+                "user1",
+                &[author_id1.to_string(), author_id2.to_string()],
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(result[author_id1].len(), 1);
+        assert_eq!(result[author_id1][0].id, expected_book_id);
+        assert!(result[author_id2].is_empty());
+    }
```
[the remark this answers is not in the document]
Comment (hiterm) [reply src/use_case/interactor/query.rs:385]: Codexとして対応しました。`e7b78d9` で `find_books_by_author_ids_rejects_invalid_id_without_calling_repository` を追加し、無効な Author ID が `Err` になることと、Repository の一括取得が0回であることを検証しています。ローカルの全163テストとGitHub Actionsの `Test Suite` も成功しています。

File: openspec/changes/add-author-books-field/tasks.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+## 3. GraphQL Resolution
+
+- [x] 3.1 Add and test `BooksByAuthorLoader` with one batched use-case call
+- [x] 3.2 Add the `Author.books` complex resolver and register its request-scoped DataLoader
+- [x] 3.3 Add GraphQL tests for populated, empty, shared, and batched author book results
+
+## 4. Verification
+
+- [x] 4.1 Run formatting, Clippy, and the complete locked test suite
```
[the remark this answers is not in the document]
Comment (hiterm) [reply openspec/changes/add-author-books-field/tasks.md:21]: Codexとして確認・対応しました。`DATABASE_URL` を設定した実 PostgreSQL 環境で、認証付き HTTP E2E テストを追加して実行し、共有 Book と空配列のケースが成功しています。push 後の GitHub Actions でも `Test Suite` と `End-to-end tests` を含む全チェックが成功したため、task 4.1 は完了のままとしています。対応コミットは `e7b78d9` です。
